Gateway/Request/CheckoutDataBuilder.php

```diff
@@ -104,12 +106,12 @@ public function build(array $buildSubject): array
             $this->paymentMethodsHelper->isOpenInvoice($paymentMethodInstance) ||
             $payment->getMethod() === AdyenPayByLinkConfigProvider::CODE
         ) {
-            if ($this->adyenHelper->isPaymentMethodOfType($payment->getMethod(), Data::KLARNA) &&
+            if (str_contains($payment->getMethod(), self::KLARNA) &&
                 $this->configHelper->getAutoCaptureOpenInvoice($storeId)) {
                 $requestBody['captureDelayHours'] = 0;
             }
 
-            if ($this->adyenHelper->isPaymentMethodOfType($payment->getMethod(), Data::KLARNA) ||
+            if (str_contains($payment->getMethod(), self::KLARNA) ||
```

Comment:
What would you think about moving the `KLARNA` constant to the `PaymentMethods` helper as we already have some method name constants store there?